### PR TITLE
JDK 11 upgrade work, prep for vector-search feature

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Branch for work to eventually support vector-search functionality once available via Cassandra 5.x but will start with changes to first support using JDK 11 as the baseline, as this is a pre-requisite to work with newer backends.

For JDK 11 changes needed:

- Update JDK targets, JDK used by CI, enforced limit
- Add javax.annotations dependency (separate module from core with JDK 9 and later)
- Comment out C-3.11 backend (soon to be removed from Stargate `main`)
- Add one or more `--add-exports` statements (or possible `--add-opens`) to open up JDK internal modules for Cassandra (first for unit and integration tests)

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
Looks like C*4.0 backend unit test fails with:

```
ERROR [main] 2023-06-14 17:59:48,531 FileUtils.java:106 - FATAL: Cassandra is unable to access required classes. This usually means it has been run without the aid of the standard startup scripts or the scripts have been edited. If this was intentional, and you are attempting to use Java 11+ you may need to add the --add-exports and --add-opens jvm options from either jvm11-server.options or jvm11-client.options
java.lang.IllegalAccessException: access to public member failed: jdk.internal.ref.Cleaner.clean[Ljava.lang.Object;@66971f6b/invokeVirtual, from org.apache.cassandra.io.util.FileUtils (unnamed module @bd8db5a)

```

and looks like we'd need to add module opens as per:

https://support.datastax.com/s/article/ERROR-FATAL-Cassandra-is-unable-to-access-required-classes-This-usually-means-it-has-been

File: coordinator/persistence-api/src/test/resources/logback-test.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tatu-at-datastax) [inline coordinator/persistence-api/src/test/resources/logback-test.xml:1]: Copied from persistence backends' test setups, to reduce noise from Netty DEBUG outpit.

**Comment by ivansenic** [comment]
Hell yeaaaa..

**Review by jeffreyscarpenter — APPROVED** [review]
Approving. We know not all ITs pass but we will merge and continue progress on follow up PR.  Not our standard practice but ok in this case for bootstrapping new feature branch.

**Comment by tatu-at-datastax** [comment]
Looks like something is trying to start Java 8 JVM which results in test failures, fwtw.

**Pull request merged**